src/app/modules/item/components/item-permissions/item-permissions.component.ts

```diff
@@ -33,6 +34,17 @@ export class ItemPermissionsComponent implements OnChanges {
       ...this.itemData.item.watchedGroup.permissions,
       canMakeSessionOfficial: false,
     } : undefined;
+
+    const currentUserCanGrantGroupAccess = this.watchedGroup?.currentUserCanGrantGroupAccess;
+    const currentUserCanGivePermissions = this.watchedGroupPermissions &&
+      (this.watchedGroupPermissions.canGrantView !== 'none' ||
+        this.watchedGroupPermissions.canWatch === 'answer_with_grant' ||
+        this.watchedGroupPermissions.canEdit === 'all_with_grant');
```

Comment:
In my opinion, this part should be a helper function `canGivePermissions(permissions)` externalized in `src/app/modules/item/helpers/item-permissions.ts`

Reply:
Fixed